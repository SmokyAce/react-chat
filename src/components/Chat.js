import React from 'react';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Button from 'material-ui/Button';
import ChatMessageList from './ChatMessageList';
import MessageInput from './MessageInput';
import InfoPaper from './InfoPaper';

const styles = theme => ({
  chatLayout: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    paddingTop: '64px',
    height: '100%',
    overflow: 'hidden',
    width: 'calc(100% - 320px)',
  },
  messageInputWrapper: {
    position: 'fixed',
    left: 'auto',
    right: 0,
    bottom: 0,
    width: 'calc(100% - 320px)',
    padding: theme.spacing.unit * 3,
  },
  messageInput: {
    padding: theme.spacing.unit * 2,
  },
});

const ChatContent = ({
  classes, messages, user, actions, activeChat, isConnected,
}) => (
  <main className={classes.chatLayout}>
    {activeChat ? (
      <React.Fragment>
        <ChatMessageList messages={messages} user={user} />
        <div className={classes.messageInputWrapper}>
          <Paper className={classes.messageInput} elevation={6}>
            {user.isCreator || user.isMember ? (
              <MessageInput sendMessage={actions.sendMessage} disabled={!isConnected} />
            ) : (
              <Button
                variant="raised"
                color="primary"
                fullWidth
                onClick={actions.joinChat}
                disabled={!isConnected}
              >
                Join
              </Button>
            )}
          </Paper>
        </div>
      </React.Fragment>
    ) : (
      <InfoPaper />
    )}
  </main>
);

export default withStyles(styles)(ChatContent);
