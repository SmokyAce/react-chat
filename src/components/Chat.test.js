/* eslint-env jest */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Chat from './Chat';

jest.mock('./ChatMessageList', () => () => 'ChatMessageList');
jest.mock('./MessageInput', () => () => 'MessageInput');

const mockProps = {
  user: {
    username: 'test',
    isMember: true,
    isCreator: true,
  },
  activeChat: {
    _id: '1',
    title: 'test',
  },
  isConnected: false,
  messages: [
    {
      chatId: '123',
      content: 'content',
      sender: {},
      createdAt: '2018-03-08T13:57:58.706Z',
    },
  ],
  actions: {
    joinChat: jest.fn(),
    isConnected: true,
    sendMessage: jest.fn(),
  },
  hasShift: false,
};

describe('<Chat />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <Chat {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  /* eslint-disable react/jsx-indent */
  it('renders correctly ', () => {
    const tree = renderer
      .create(<MemoryRouter>
          <Chat {...mockProps} />
              </MemoryRouter>)
      .toJSON();
    /* eslint-enable react/jsx-indent */

    expect(tree).toMatchSnapshot();
  });
});
