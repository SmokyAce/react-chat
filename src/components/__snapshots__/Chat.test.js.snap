// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> renders correctly  1`] = `
<main
  className="Chat-content-33"
>
  ChatMessageList
  <div
    className="Chat-messageInputWrapper-35"
  >
    <div
      className="MuiPaper-root-38 MuiPaper-shadow6-46 MuiPaper-rounded-39 Chat-messageInput-36"
    >
      MessageInput
    </div>
  </div>
</main>
`;
